modules/gke/iam.tf

```diff
@@ -1,7 +1,4 @@
 resource "google_service_account_iam_binding" "spacelift-backend_workload-identity" {
-  # The cluster creates the serviceAccount, so we need to wait until it finishes
-  depends_on = [google_container_cluster.spacelift]
```

Comment:
not having this completely broke the deployment for me iirc.

Reply:
Because it was working fine for me, and that is actually incorrect to say that the cluster creates the `serviceAccount`.
I tried it with disabling GKE entirely, and the binding was created fine.

I need to double check.

Reply:
I can't reproduce any issue, I disabled the creation of the cluster and applied. And the role is granted correctly.

![image](https://github.com/user-attachments/assets/c54b26eb-30ee-42db-a150-f6eb33722998)

Reply:
cool!